packages/core/src/Step.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import Flex from './Flex'
+import Text from './Text'
+import CheckIcon from 'pcln-icons/lib/Check'
+
+const StyledFlex = styled(Flex)`
+  cursor: ${props => (props.onClick ? 'pointer' : 'default')};
+`
+
+function Step({ className, active, completed, children, onClick }) {
+  const color = active || completed ? 'primary' : 'text.light'
+
+  return (
+    <StyledFlex className={className} alignItems="center" onClick={onClick}>
+      {completed && (
+        <CheckIcon color="primary" size={16} mr={1} legacy={false} />
```

Comment:
Considering this Icon is not purely decorative and it provides context to the state of the current state, we should add an `aria-label="Step Completed"` or make the label dynamic if needed, so that it can be passed as a prop. This is needed so that when Screen Reader users tab onto this item, they know that it has been completed.

Reply:
- [ ] add an aria-label="Step Completed" or make the label dynamic if needed